### ISSUE TITLE
Once Human

### ISSUE THREAD

**Issue by glenni83** [doitsujin/dxvk#4499]
There are some major rendering issues in this game. Its been like this for months. It used to play pretty ok on proton-ge 7.55, with a few graphical errors. Then the game got an anti cheat, and we could not use it anymore. It did never work on newer protons. I believe that the proton team has also look at this issue for months now, without a solution.

So some places are rendering pretty good, like the torturial etc. After you get into the open world, every npc, player house, animals starts to stutter heavy. It gets so bad, you get motion sick. Moving and driving is pretty bad in general. 

From the look of things, it looks like the whole community has turned upside down on this game, to make it work. 

Computer Information:
Manufacturer: ASUSTeK COMPUTER INC.
Model: ROG STRIX X670E-E GAMING WIFI
Form Factor: Desktop
No Touch Input Detected
Processor Information:
CPU Vendor: AuthenticAMD
CPU Brand: AMD Ryzen 7 7800X3D 8-Core Processor
CPU Family: 0x19
CPU Model: 0x61
CPU Stepping: 0x2
CPU Type: 0x0
Speed: 5050 MHz
8 logical processors
8 physical processors
Hyper-threading: Unsupported
FCMOV: Supported
SSE2: Supported
SSE3: Supported
SSSE3: Supported
SSE4a: Supported
SSE41: Supported
SSE42: Supported
AES: Supported
AVX: Supported
AVX2: Supported
AVX512F: Supported
AVX512PF: Unsupported
AVX512ER: Unsupported
AVX512CD: Supported
AVX512VNNI: Supported
SHA: Supported
CMPXCHG16B: Supported
LAHF/SAHF: Supported
PrefetchW: Unsupported
BMI1: Supported
BMI2: Supported
F16C: Supported
FMA: Supported
Operating System Version:
"CachyOS" (64 bit)
Kernel Name: Linux
Kernel Version: 6.12.1-2-cachyos
X Server Vendor: The X.Org Foundation
X Server Release: 12401004
X Window Manager: KWin
Steam Runtime Version: <Runtime disabled>
Client Information:
Version: 1731433018
Browser GPU Acceleration Status: Disabled
Browser Canvas: Unavailable
Browser Canvas out-of-process rasterization: Disabled
Browser Direct Rendering Display Compositor: Disabled
Browser Compositing: Disabled
Browser Multiple Raster Threads: Enabled
Browser OpenGL: Disabled
Browser Rasterization: Disabled
Browser Raw Draw: Disabled
Browser Skia Graphite: Disabled
Browser Video Decode: Disabled
Browser Video Encode: Disabled
Browser Vulkan: Disabled
Browser WebGL: Unavailable
Browser WebGL2: Unavailable
Browser WebGPU: Disabled
Browser WebNN: Disabled
Video Card:
Driver: NVIDIA Corporation NVIDIA GeForce RTX 4090/PCIe/SSE2
Driver Version: 4.6.0 NVIDIA 565.57.01
Desktop Color Depth: 24 bits per pixel
Monitor Refresh Rate: 137 Hz
VendorID: 0x10de
DeviceID: 0x2684
Revision Not Detected
Number of Monitors: 2
Number of Logical Video Cards: 1
Primary Display Resolution: 3840 x 2160
Desktop Resolution: 6400 x 2160
Primary Display Size: 36.22" x 20.35" (41.54" diag), 92.0cm x 51.7cm (105.5cm diag)
Primary VRAM: 24564 MB
Sound card:
Audio device: Nvidia GPU a3 HDMI/DP
Memory:
RAM: 31722 Mb
VR Hardware:
VR Headset: None detected
Miscellaneous:
UI Language: English
LANG: en_GB.UTF-8
Total Hard Disk Space Available: 1181942 MB
Largest Free Hard Disk Block: 941491 MB
Storage:
Number of SSDs: 5
SSD sizes: 4000G,4000G,3840G,250G,0B
Number of HDDs: 0
Number of removable drives: 0

https://filebin.net/owph8spy8wppn5pd)

https://www.protondb.com/app/2139460

**Comment by Blisto91** [comment]
Hi there.
Does the game doesn't work on Linux at all right now because of anti cheat? If so then it is likely pretty hard to look in to. Even if when taking a look at windows anti cheats often don't allow the kind of tools used to debug visual issues (with good reason).

**Comment by Blisto91** [comment]
Because of the visual issues right? Not anti cheat.

I will try to take a look here later.

**Comment by glenni83** [comment]
The game is free on steam, it made by NetEase with the NeoX engine. Its fully working on linux. just not playable due to stutter and bad performance in game. You can look at the ProtonDB to look at all the issues. 

I run the game on a 4090, on very low settings, just then its pretty much not playable.

**Comment by Blisto91** [comment]
Looking at your log it seems to be hinting that the game uses d3d12 to render and not d3d11.  That would make it more appropriate to make a issue at https://github.com/HansKristian-Work/vkd3d-proton

But a bell is ringing for this game now. Could you try to installed the native version of d3dcompiler_47 with protontricks and see how it behaves then?

**Comment by glenni83** [comment]
> Looking at your log it seems to be hinting that the game uses d3d12 to render and not d3d11. That would make it more appropriate to make a issue at https://github.com/HansKristian-Work/vkd3d-proton
> 
> But a bell is ringing for this game now. Could you try to install the native version of d3dcompiler_47 with protontricks and see how it behaves then?

It has both. d3d11 does not work at all right now. If you go into that, all textures break

**Comment by Blisto91** [comment]
Could you check if either d3d11 or d3d12 mode gets fixed with a native d3dcompiler_47?

**Comment by glenni83** [comment]
I did copy this post over at Hans Kristian as well. Both are broken of the versions.

**Comment by glenni83** [comment]
![image](https://github.com/user-attachments/assets/fde04f55-f47d-4ba9-ad27-049906d45a5e)
![rn_image_picker_lib_temp_20a82c8a-59a9-4a47-97ec-14ef3b7ca80f](https://github.com/user-attachments/assets/b3b5aab1-8c8f-435e-b557-4a941de480b5)

**Comment by glenni83** [comment]
> So i have reproduced the visual glitches in dx11 mode out of the box yes, but after i start using the Windows d3dcompiler_47 it appears normal without glitches. This is tested with Experimental Bleeding Edge.

Im aware of that. But it still dont fix the performance issues.  So im not sure how to solve it. Or where to go with this problem.

**Comment by Blisto91** [comment]
I think i need to start over to properly understand what this issue is about.
The performance/stutter issue you are talking about is that both in dx11 and dx12 mode?

**Comment by glenni83** [comment]
> I think i need to start over to properly understand what this issue is about. Most of the opening post seem to write about the visual issue. The performance/stutter issue you are talking about is that both in dx11 and dx12 mode?

Yes the stutter is on both. To be honest, im not even sure anymore what is causing it. Other that objects, npc, and building causes major stutter. To me it seems to be a render issue of some sort. Been turning upside down on this for months. Then I tried to compare what was the difference from the proton 7 to the once we have now. Proton 7, was the only ok working proton for this. Any newer or older was not running good at all. Then a few months ago, the game got an anti cheat. Meaning we could not use proton 7. Due to its not compatibel with anti cheat. I dont think the anti cheat it self has anything to do with it. It ran close to flawless on the proton-ge 7-55. So nobody understands why.

**Comment by Shoun2137** [comment]
Specs:
R5 1600AF, 16GB RAM, GTX 1060 6GB, ArtixLinux w/ 570.133.07 proprietary novideo drivers, Pure Steam and Proton Experimental or Proton Hotfix launches the game fine.

Game works pretty good when it comes to performance with occasional stuttering (prolly due to shader compilation) HOWEVER, there is severe shader artifacting (those orange outlines in some shader) that is probably caused by drivers as this stuff was reported to happen on VKD3D and Windows (Or it's just the game that's broken):
https://old.reddit.com/r/OnceHumanOfficial/comments/1j9s99k/i_need_help_with_graphics_settings_please/
https://old.reddit.com/r/OnceHumanOfficial/comments/1j6l2og/help_me_how_do_i_fix_this_orange_outline/

And those screenshots are mine:
https://i.imgur.com/nreOvTF.jpeg
https://i.imgur.com/r1fiJLp.jpeg

Dropping this just to update the thread with some fresh info. Also there are people that succesfully play the game (albeit with VKD3D which doesn't fix the orange outlines for me sadly): https://www.youtube.com/watch?v=c4UgD3uV2AA
